src/ui/ButtonIcon/ButtonIcon.tsx

```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+
+import styles from './ButtonIcon.module.css';
+
+interface ButtonIconProps {
+    src: string;
+    alt: string;
+    buttonProps?: React.ComponentProps<'button'>;
+}
+
+function ButtonIcon({ src, alt, ...buttonProps }: ButtonIconProps) {
+    return (
+        <button className={styles.button} {...buttonProps}>
+            <img src={src} alt={alt} />
+        </button>
+    );
+}
+
+export default ButtonIcon;
```

Comment:
if u add new props to your interface, and forget destructured it, you will get a warning in. browser console, so my suggestion is replace interface with type, and create a union of your props and native button props.

Also suggest u do not use interfasec use types in TS they are more powerfull and flexible
interface using when u want to extend some interfaces of external libraries

```suggestion
type ButtonIconProps = React.ComponentProps<'button'> & {
    src: string;
    alt: string;
}

function ButtonIcon({ src, alt, ...buttonProps }: ButtonIconProps) {
    return (
        <button className={styles.button} {...buttonProps}>
            <img src={src} alt={alt} />
        </button>
    );
}

export default ButtonIcon;
```

Reply:
I didn't use this custom component. I just forgot delete him, but thank you for good suggestion

Reply:
Deleted